### ISSUE TITLE
aurelia-i18n and checkbox translation not working

### ISSUE THREAD

**Issue by mar9000** [aurelia-ui-toolkits/aurelia-materialize-bridge#305]
I've tried translating checkbox label using what the i18n API suggested:
```html
<md-checkbox t="rememberMe" md-checked.bind="rememberMe">R</md-checkbox>
```
the string "Remember me" gets translated but the widget losts its `input` and `label` elements resulting in such HTML fragment:
```html
<md-checkbox t="rememberMe" md-checked.bind="rememberMe"
             class="au-target" au-target-id="97">Remember me</md-checkbox>
```
This is because i18n.updateValue(), if `attr` == 'text', runs through this code:
```javascript
     node._newChild = newChild;
     while (node.firstChild) {
          node.removeChild(node.firstChild);
     }
```
That removes from the generated `md-checkbox` content the `input` and `label` elements.
I've tested this solution that however it's probably far from optimal: `checkbox.html` should use a named slot:
```html
<template>
  <input type="checkbox" id="${controlId}" ref="checkbox" blur.trigger="blur()" />
  <label for="${controlId}">
    <slot name="label"></slot>
  </label>
</template>
```
then its usage could be:
```html
<md-checkbox md-checked.bind="rememberMe"><span t="rememberMe" slot="label">RS</span></md-checkbox>
```
With my actual knowledge I can't say where the fix should go, in `checkbox.html` in `i18n.js` or in both. The problem here is that the actual format for inserting checkbox stops working, so the following results in an empty label:
```html
<md-checkbox md-checked.bind="rememberMe">Remember me</md-checkbox>
```

**Comment by Thanood** [comment]
```javascript
while (node.firstChild) {
  node.removeChild(node.firstChild);
}
```

If `node` is the `md-checkbox` element, why would that not also remove the slot?
I'm currently setting up a skeleton with which I can try this. If you already have anything I could clone then this would help. :smiley:

**Comment by mar9000** [comment]
In fact removing the label it removes also the slot it contains and substitutes everything with a text element created here:
```javascript
newChild = _aureliaPal.DOM.createTextNode(this.tr(key, params));
```
About skeleton: what do you think if I try to add a i18n section to the component browser? So in the future one can post non working i18n fragment that can be dropped into that section.

**Comment by Thanood** [comment]
> About skeleton: what do you think if I try to add a i18n section to the component browser?

I like the idea. :+1: 
I just think because it doesn't work it wouldn't help people much. :smiley: 
An i18n section would be great, anyway. So, yes.

Also probably worth trying to put i18n into a gist.run..

I must confess I'm so "virgin" regarding aurelia-i18n that I still have to do the basics from their readme right now. :smiley:

**Comment by Thanood** [comment]
I must have been blind. Because...

```html
<md-checkbox md-checked.bind="rememberMe"><span t="rememberMe">R</span></md-checkbox>
```

Seems to work. :smiley:

**Comment by Thanood** [comment]
What do you think about translating the whole catalog? I mean in principle, not that it's necessary you yourself who does it (although, of course, I'd appreciate this). :smiley: 

I can provide German translation and I could probably get Dutch. Maybe Russian, too..

**Comment by mar9000** [comment]
Ok the snippet you've posted is working, I haven't tried it. However it's tricky, I mean without translation I can remove the `span` element with translation I cannot translate the md-checkbox content but add a span and translate it. Should probably ask to the i18n guys if this is an intendent use case. 
But ok I'll close this issue, we will speak again about it when a translated section of the component catalog will be in place.

**Comment by Thanood** [comment]
I guess they could have used `textContent` or `innerText` but I know that this is much slower (at least `innerText` on IE).

**Comment by mar9000** [comment]
After a quick look at the component browser I think I could
  * add the language selector beside the color-switcher
  * modify app.js and main.js to load i18n and set default language to english
  * add some working examples for instance label, input and buttons. For each component it's probably advisable to add a section *aurelia-i18n integration*.
  * into the *aurelia-i18n integration* one can writes working and not working examples.
what do you think?

**Comment by Thanood** [comment]
Good idea on the language selector. :+1: 
So, this means I can see i18n in action by switching to Italian and in each sample there's a section "i18n" which shows the implementation for the current component?

This is good.
The thing is this (not really a problem, therefore "thing"):
* The component browser shows the *actual source* of the current page (although not all examples are migrated to this format).
* That means, i18n is already in each sample.
* Which does not mean it's bad. :smiley: 

The i18n section could still carry some more information about how to achieve i18n.
I don't want the samples to be too "overwhelming" but when I get this right, actual *usage*  of i18n is just an additional attribute, so that's easy.

TLDR: Let's try what it looks like.. :smiley:
